### PR TITLE
First version of ProCoSysSettings to handle all settings

### PR DESCRIPTION
**Pull request by paal123**
First version of ProCoSysSettings, currently only replacing the direct read of settings.json

Preparing for feature flags, but no implementation yet.

### CODE REVIEW

File: src/auth/AuthService.tsx [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
-        authority: settings.auth.authority,
+        authority: ProCoSysSettings.auth.authority,
     },
     // system: {
     //     logger: new Logger((lvl: any, message: any, piEnabled?: boolean ): void => { console.log('Auth: ', message);})
     // }
 };
 
-const defaultLoginScopes = JSON.parse(settings.auth.defaultScopes.replace(/'/g, '"'));
+const defaultLoginScopes = JSON.parse(ProCoSysSettings.auth.defaultScopes.replace(/'/g, '"'));
```
Comment (Cnordbo) [inline src/auth/AuthService.tsx:17]: Add this parsing directly to `ProCoSysSettings` static property, so thats its ready to use.

**Review by Cnordbo — CHANGES_REQUESTED** [review]
A few minor details, can be merged when considered or fixed.

File: src/auth/AuthService.tsx (hunk 1 above)
Comment (paal123) [reply src/auth/AuthService.tsx:17]: Agree!

File: settings.template.json [hunk 2]
```diff
@@ -4,6 +4,13 @@
         "authority": "",
         "defaultScopes": "[]"
     },
+    "features": {
+        "ipoEnabled": "false",
+        "libraryEnabled": "true",
+        "preserverationEnabled": "true",
+        "graphEnabled": "true",
+        "mainEnabled": "true"
+    },
```
[the remark this answers is not in the document]
Comment (paal123) [reply settings.template.json:13]: Agree!

**Pull request merged**